### PR TITLE
Publish Codellama instruct models

### PR DESCRIPTION
**Pull request by vikasagrawal-ms**
Publish Codellama instruct models 7b 13b 34b
## Problem
Add code-llama base models.

## Steps
1. Converted model to mlflow using hftransformers flavor
2. Used transformers 4.31.1 to create TGI env for online deployment

### Model Details ###

**MLModel**

```
flavors:
  hftransformersv2:
    code: null
    generator_config:
      do_sample: true
      max_new_tokens: 256
    model_hf_load_kwargs:
      device_map: auto
      torch_dtype: torch.bfloat16
    hf_config_class: LlamaConfig
    hf_pretrained_class: LlamaForCausalLM
    hf_tokenizer_class: AutoTokenizer
    model_data: data
    pytorch_version: 2.0.0+cu117
    task_type: text-generation
    transformers_version: 4.33.1
  python_function:
    data: data
    env: conda.yaml
    loader_module: azureml.evaluate.mlflow.hftransformers
    python_version: 3.8.17
metadata:
  azureml.base_image: mcr.microsoft.com/azureml/curated/foundation-model-inference:12
mlflow_version: 2.3.1
model_uuid: 656417aee41f499a885419e4bd9cb14c
utc_time_created: '2023-08-30 10:08:57.091028'
```

**conda.yaml**

```
channels:
- conda-forge
dependencies:
- python=3.8.17
- git
- pip<=23.2.1
- pip:
  - mlflow==2.3.1
  - cloudpickle==2.2.1
  - jsonpickle==3.0.2
  - mlflow-skinny==2.3.1
  - azureml-core==1.53.0
  - azureml-mlflow==1.53.0
  - azureml-metrics==0.0.25.post1
  - scikit-learn==1.1.0
  - cryptography==41.0.3
  - python-dateutil==2.8.2
  - datasets==2.14.4
  - soundfile==0.12.1
  - librosa==0.10.1
  - diffusers==0.20.1
  - sentencepiece==0.1.99
  - transformers==4.33.1
  - torch==2.0.0
  - accelerate==0.22.0
  - Pillow==9.4.0
  - azureml-evaluate-mlflow==0.0.25
- pycocotools=2.0.4
name: mlflow-env
```

## Test
1. Successfully validated TGI deployment for all models and collected response which is part inference sample in respective model description 
4. Successfully validated batch deployment with existing models in test registry

**34b** 
* Online Endpoint: [link](https://ml.azure.com/endpoints/realtime/tgi-codellama-34b-instruct-hf-1/test?wsid=/subscriptions/b17253fa-f327-42d6-9686-f3e553e24763/resourceGroups/2424-rg/providers/Microsoft.MachineLearningServices/workspaces/model-validations-eastus&tid=72f988bf-86f1-41af-91ab-2d7cd011db47)

image

* batch endpoint run: [link](https://ml.azure.com/experiments/id/8924b78d-9c9b-48f4-8874-ce67e34029e7/runs/batchjob-1a695142-dead-44e5-b4af-f2b40ef95b1b?wsid=/subscriptions/b17253fa-f327-42d6-9686-f3e553e24763/resourceGroups/2424-rg/providers/Microsoft.MachineLearningServices/workspaces/model-validations-eastus&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#)

**13b**
* Online Endpoint: [link](https://ml.azure.com/endpoints/realtime/tgi-codellama-13b-instruct-hf-1/test?wsid=/subscriptions/ed2cab61-14cc-4fb3-ac23-d72609214cfd/resourceGroups/donotuseordelete-runner-rg/providers/Microsoft.MachineLearningServices/workspaces/runners-workspace&tid=72f988bf-86f1-41af-91ab-2d7cd011db47)

image

* batch endpoint run: [link](https://ml.azure.com/experiments/id/1c07e01b-f38a-4fc4-9343-6346568235c1/runs/batchjob-2f50d35b-18a0-4f02-bc2f-b0f09b24ca18?wsid=/subscriptions/ed2cab61-14cc-4fb3-ac23-d72609214cfd/resourceGroups/donotuseordelete-runner-rg/providers/Microsoft.MachineLearningServices/workspaces/runners-workspace&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#)

**7b**
* Online Endpoint: [link](https://ml.azure.com/endpoints/realtime/tgi-codellama-7b-instruct-hf-202/test?wsid=/subscriptions/72c03bf3-4e69-41af-9532-dfcdc3eefef4/resourceGroups/shared-finetuning-rg/providers/Microsoft.MachineLearningServices/workspaces/model-mgmt-australiaeast&tid=72f988bf-86f1-41af-91ab-2d7cd011db47)

image

* batch endpoint run: [link](https://ml.azure.com/runs/batchjob-08941af1-9ed5-4741-8191-01b91ca97ed9?wsid=/subscriptions/72c03bf3-4e69-41af-9532-dfcdc3eefef4/resourceGroups/shared-finetuning-rg/providers/Microsoft.MachineLearningServices/workspaces/model-mgmt-australiaeast&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#)

### CODE REVIEW

File: assets/models/system/codellama-34b-instruct-hf/spec.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  - hf_tgi
+  evaluation_compute_allow_list:
+  - Standard_ND40rs_v2
+  - Standard_ND96asr_v4
+  - Standard_ND96amsr_v4
+  inference_compute_allow_list:
+  - Standard_ND96asr_v4
+  - Standard_ND96amsr_v4
+  finetune_compute_allow_list:
+  - Standard_ND96amsr_A100_v4
```
Comment (skasturi) [inline assets/models/system/codellama-34b-instruct-hf/spec.yaml:22]: Could we please confirm with @amangupta26 that FT can be enabled?

**Review by skasturi — DISMISSED** [review]
Let us ensure FT, Eval pass. If not, we can revert by removing those configurations

**Comment by skasturi** [comment]
please wait for final nnames from @swatig007

File: assets/models/system/codellama-34b-instruct-hf/spec.yaml (hunk 1 above)
Comment (vikasagrawal-ms) [reply assets/models/system/codellama-34b-instruct-hf/spec.yaml]: Removed

**Comment by vikasagrawal-ms** [comment]
> Let us ensure FT, Eval pass. If not, we can revert by removing those configurations

removed those configurations

**Review by HrishikeshGeedMS — APPROVED** [review]

**Review by novaturient95 — APPROVED** [review]

**Pull request merged**